R/circa_single.R

```diff
@@ -81,6 +91,14 @@ circa_single <- function(x,
       ))
     }
   }
+  
+  if(!is.null(sample_weights)){
+    l.weights <- length(sample_weights)
```

Comment:
Could you please change this to `snake_case` to be consistent with the rest of the package? i.e: `l.weights` to become `len_weights`.

Reply:
done, and also moved the weights validation from the `circa*.R` scripts to `utils.R` to avoid redundancy. Now each `circa*.R` calls `check_weights()` to validate that the weights have same length as the data, are non-negative, numeric and do not contain NAs.